webapp/channels/src/components/user_settings/security/index.ts

```diff
@@ -35,7 +35,7 @@ function mapStateToProps(state: GlobalState, ownProps: Props) {
     const userHasTokenRole = UserUtils.hasUserAccessTokenRole(ownProps.user.roles) || UserUtils.isSystemAdmin(ownProps.user.roles);
 
     const enableOAuthServiceProvider = config.EnableOAuthServiceProvider === 'true';
-    const enableSignUpWithEmail = config.EnableSignUpWithEmail === 'true';
+    const allowedToSwitchToEmail = config.EnableSignInWithEmail === 'true' && (config.EnableSignInWithEmail === 'true' || config.EnableSignInWithUsername === 'true');
```

Comment:
Should `config.EnableSignInWithEmail === 'true' && ....` be `config.EnableSignUpWithEmail === 'true' &&....`?

Reply:
Excellent catch, thanks @BenCookie95 :100: